app/jobs/membership_event_job.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+# Documentation: https://developer.github.com/v3/activity/events/types/#membershipevent
+class MembershipEventJob < ApplicationJob
+  queue_as :github_event
+
+  def perform(payload_body)
+    return true unless payload_body.dig("action") == "removed"
+
+    user_id = payload_body.dig("member", "id")
+    team_id = payload_body.dig("team", "id")
+    user = User.find_by(uid: user_id)
+    group_id = Group.find_by(github_team_id: team_id).id
+
+    repo_accesses = GroupAssignmentRepo.find_by(group_id: group_id).repo_accesses
```

Comment:
So `Group` has `repo_accesses` directly https://github.com/education/classroom/blob/44efed0852076877840d2dcd387389f165e0eaa5/app/models/group.rb#L13-L14

So you could just do:

```ruby
def perform(payload_body)
  return true unless payload_body.dig("action") == "removed"
  
  user_id = payload_body.dig("member", "id")
  team_id = payload_body.dig("team", "id")

  user  = User.find_by(uid: user_id)
  return true unless user.present?

  group = Group.find_by(github_team_id: team_id)
  return true unless group.present?

  group.repo_accesses.find_by(user_id: user.id)
  repo_access&.destroy
end
```

Reply:
I get it now thanks ! 👍